wheel/pep425tags.py

```diff
@@ -115,15 +115,15 @@ def calculate_macosx_platform_tag(archive_root, platform_tag):
     """
     prefix, base_version, suffix = platform_tag.split('-')
     base_version = tuple([int(x) for x in base_version.split(".")])
-    if len(base_version) == 2:
-        base_version = base_version + (0,)
+    if len(base_version) >= 2:
+        base_version = base_version[0:2]
 
-    assert len(base_version) == 3
+    assert len(base_version) == 2
     if "MACOSX_DEPLOYMENT_TARGET" in os.environ:
         deploy_target = tuple([int(x) for x in os.environ[
             "MACOSX_DEPLOYMENT_TARGET"].split(".")])
-        if len(deploy_target) == 2:
-            deploy_target = deploy_target + (0,)
+        if len(deploy_target) >= 2:
+            deploy_target = deploy_target[0:2]
```

Comment:
So the information we lose here is not important? There's never an important difference between e.g. `10.14.0` and `10.14.2`?

Reply:
So I'm wondering if we should work with full 3-part versions to make all checks, and only cut off the last version later, when creating the tag string?

Reply:
I think that this option is only valid with major and a minor version numbers

Apologies for old docs, it's all I can find atm 
https://developer.apple.com/library/archive/documentation/DeveloperTools/Reference/XcodeBuildSettingRef/1-Build_Setting_Reference/build_setting_ref.html#//apple_ref/doc/uid/TP40003931-CH3-SW129

Reply:
OK, yes, the documentation on this is so crappy that except for that old version of Apple's docs, all I can seem to find is a bunch of SO questions and issues/PRs on GitHub :-|

I was mainly looking at https://stackoverflow.com/a/25362535 where `10.3.9` is mentioned, but I can't find that anywhere else (also tried the man page on a mac I have access to; doesn't show the same message anymore).

It's also quite funny that the versions in the `dylib`s are stored as `x.y.z`, then? (cfr. https://github.com/pypa/wheel/blob/fcd78f3c04c4031a0563d1b9c129eaefa83a8698/wheel/macosx_libfile.py#L345-L350)


At any rate, maybe the value from the environment variable should be checked, then, to make sure it only contains 2 or 3 parts and not more or less? This is some for of user input, after all.

Reply:
Whether or not the bugfix might contain relevant information, the [platform_tag in Python](https://github.com/pypa/packaging/blob/master/packaging/tags.py#L379) considers all bugfix releases within the same minor release of macOS as equivalent.

I tend to agree with this, FWIW. I ran a regex on the macOS 10.15 SDK and I can't find any APIs that specify a macOS version to the bugfix release.

Speculating, I think that sometimes Apple do bugfix releases to target new hardware, and those would require such a detailed version string. But I don't think it's used when building against the public SDK.

Reply:
> Whether or not the bugfix might contain relevant information, the [platform_tag in Python](https://github.com/pypa/packaging/blob/master/packaging/tags.py#L379) considers all bugfix releases within the same minor release of macOS as equivalent.

OK, nice find! Better ignore it then, and have a bit of consistency :-)